### PR TITLE
[charts] Allow `SeriesValueFormatter` to return `null` value

### PR DESCRIPTION
**Pull request by clins1994**
# TL;DR

A. Omit irrelevant tooltip values by updating the SeriesValueFormatter type to allow retuning `null` values
B. Usage: update `valueFormatter` function to return null if the value is irrelevant

# Problem

Use case: when displaying multiple series with the same label/color on a stacked bar chart it can be useful to omit the series that have no value for a given stack.

Note: I am interested in seeing 'axis' tooltip not 'item'

```
const series = [
  { ... , valueFormatter: (v: number | null) => v ? `${v} 分` : '-' }
]
```

The screenshot below shows the tooltip of a stack that has "MOVING", "STOPPED", and "MOVING"

In this example the order of these events matter both on the x axis and y axis so it was decided to split the multiple "MOVING" data into separate series

When hovering over the tooltip it becomes polluted with irrelevant '-' values.

Screenshot 2024-10-22 at 19 22 50

I am able to filter out the values by passing `null` as a return but I get a type error which I was able to ignore using `!`

```
const series = [
  { ... , valueFormatter: (v: number | null) => v ? `${v} 分` : null! }
]
```

With this I can visualize only the data that is displayed on the chart

Screenshot 2024-10-22 at 19 22 16

# Solution

Since forcing the return of `null` didn't break but actually fixed my issue I wrote this PR to update the type to match the already allowed and expected behavior

### CODE REVIEW

**Comment by JCQuintas** [comment]
Thanks @clins1994, this seems to be a good addition. @alexfauquette do you see any issue?

**Review by JCQuintas — APPROVED** [review]

**Review by alexfauquette — APPROVED** [review]
LGTM 👍

I considered moving this to the fact that the `value` is `null` instead of the formatted one. But a user could have the opposite need: showing something in the tooltip for missing values
